six/modules/c++/six.sidd/tests/test_compare_sidd.cpp

```diff
@@ -0,0 +1,208 @@
+#include <import/cli.h>
+#include <import/sys.h>
+#include <import/mem.h>
+#include <import/except.h>
+
+#include <import/six.h>
+#include <import/six/sicd.h>
+#include <import/six/sidd.h>
+
+#include <cstring>
+
+mem::SharedPtr<six::Data> readMetadata(six::NITFReadControl& reader)
+{
+    const six::Data* data = reader.getContainer()->getData(0);
+    mem::SharedPtr<six::Data> retv(data->clone());
+    return retv;
+}
+
+void readWideband(six::NITFReadControl& reader, 
+        size_t numRows, 
+        size_t numCols,
+        size_t bytesPerElement,
+        mem::ScopedAlignedArray<sys::Int8_T>& buffer)
+{
+    buffer.reset( numRows * numCols * bytesPerElement );
+    six::Region region;
+    region.setStartRow(0);
+    region.setStartCol(0);
+    region.setNumRows(numRows);
+    region.setNumCols(numCols);
+    region.setBuffer(reinterpret_cast<six::UByte*>(buffer.get()));
+    reader.interleaved(region, 0);
+}
+
+bool siddsMatch(const std::string& sidd1Path, 
+        const std::string& sidd2Path, 
+        bool ignoreMetadata,
+        bool ignoreDate)
+{
+    six::XMLControlRegistry xmlRegistry;
+    xmlRegistry.addCreator(six::DataType::COMPLEX,
+            new six::XMLControlCreatorT<
+            six::sicd::ComplexXMLControl>());
+    xmlRegistry.addCreator(six::DataType::DERIVED,
+            new six::XMLControlCreatorT<
+            six::sidd::DerivedXMLControl>()); 
+
+    six::NITFReadControl reader;
+    reader.setXMLControlRegistry(&xmlRegistry);
+
+    reader.load(sidd1Path);
+    mem::SharedPtr<six::Data> sidd1Metadata = readMetadata(reader);
+    mem::ScopedAlignedArray<sys::Int8_T> sidd1Buffer;
+    readWideband(reader,
```

Comment:
You could simplify this `readWideband()` function if you just took in a const ref to a `Data` object - then you could take in just one param for that rather than these three